### DISCUSSION TITLE
Checkbox does not render the "check" mark... sometimes

### DISCUSSION THREAD

**Discussion by nesquerra** [twilio-labs/paste#2374] [General] [answered]
This might be some kind of weird rendering issue, but we've come across a situation where the checkbox only renders the blue background (using the standard theme). I haven't been able to reproduce the issue, but we have seen it from time to time...

Browser seems to be the latest and no custom themes/plugins have been added to the browser. Not sure there's much y'all can do about this, but thought I'd bring it up. Checkboxes are still usable and work.

Edit: When you uncheck a checked box (checked by default) then select it again, the check mark appears. So, it might only be occurring on initial render?

![example-checkbox-issue](https://user-images.githubusercontent.com/100308430/165813484-bbb09302-eba1-45ef-ac41-c09b629e75fe.png)

**Comment by TheSisb** [comment, marked as answer]
Hi @nesquerra,

Can you share the code that renders these checkboxes please? Just the portion of the render with the `<Checkbox ...` JSX and its props. 

Also, which version of paste/core are you using?

Thanks!

**Reply by nesquerra** [reply]
Soooooo, this is using the `react-hook-form-paste` via https://github.com/vankhoawin/react-hook-form-paste (former Numbers engineer created the package)

Render code is here:
https://code.hq.twilio.com/numbers/numbers-toolbox/blob/master/packages/frontend/src/domains/PhoneNumbers/AvailablePhoneNumbers/ConsoleAvailablePhoneNumberTable/components/TableBasicFilterSection.tsx#L37-L81

Called (via memo) here:
https://code.hq.twilio.com/numbers/numbers-toolbox/blob/master/packages/frontend/src/domains/PhoneNumbers/AvailablePhoneNumbers/ConsoleAvailablePhoneNumberTable/components/TableBasicFilterSection.tsx#L95

**Reply by nkrantz** [reply]
Hi @nesquerra! Looks like you're not running the most recent version of Paste, and we made some recent changes to checkbox to fix check-rendering bugs. Could you try upgrading core and letting us know if that helps?

**Reply by nesquerra** [reply]
Ahhh very cool, do y'all recommend we utilize the react-hook-form-paste library? If so, should we fork and manage it ourselves? Not that Van doesn't manage it, but he might not always be available to merge PRs.

**Reply by TheSisb** [reply]
If you're using a newer version of react-hook-form (>= v7) then react-hook-form-paste is not really necessary anymore.

**Reply by nesquerra** [reply]
Hmmmm seems to be some conflicting reports :p 
```
With the advent of react-hook-form v7, react-hook-form-paste is mostly unnecessary
```
Or would the latest paste-core + react-hook-form v7 be enough to migrate away from react-hook-form-paste?

**Reply by TheSisb** [reply]
I think we're both saying the same thing.

Said another way, if you upgrade to `react-hook-form v7` you don't need `react-hook-form-paste`. Keeping paste-core up to date is always a wise idea, also :)
